File: 프로그래머스/lv0/120886. A로 B 만들기/A로 B 만들기.js
function solution(before, after) {
    var answer = 1;
    let sH = new Map();
    for(let x of before){
        if(sH.has(x)) sH.set(x, sH.get(x)+1);
        else sH.set(x,1);
    }
    
    for(let x of after){
        if(!sH.has(x) || sH.get(x) === 0) return 0;
        sH.set(x,sH.get(x) -1);
    }
    return answer;
}

File: 프로그래머스/lv0/120886. A로 B 만들기/A로 B 만들기.py
def solution(before, after):
    answer = 0
    list_b = list(before)
    list_a = list(after)
    for i in list_b:
        if i in list_a:
            list_a.remove(i)
    return 1 if len(list_a) == 0 else 0